crates/proto/go/src/detect.rs

```diff
@@ -0,0 +1,47 @@
+use crate::GoLanguage;
+use proto_core::{async_trait, Detector, ProtoError};
+use std::fs::File;
+use std::io::{BufRead, BufReader};
+use std::path::Path;
+
+static GOPREFIX: &str = "go ";
+
+#[async_trait]
+impl Detector<'_> for GoLanguage {
+    async fn detect_version_from(&self, working_dir: &Path) -> Result<Option<String>, ProtoError> {
+        let gowork = working_dir.join("go.work");
+
+        if gowork.exists() {
+            return Ok(Some(scan_for_go_version(&gowork)?));
+        }
+
+        let gomod = working_dir.join("go.mod");
+
+        if gomod.exists() {
+            return Ok(Some(scan_for_go_version(&gomod)?));
+        }
+
+        Ok(None)
+    }
+}
+
+fn scan_for_go_version(path: &Path) -> Result<String, ProtoError> {
```

Comment:
instead of erroring and crashing the program (not super important), we should probably have this return an `Option<String>`

Reply:
Done